vaadin-testbench-shared/src/main/java/com/vaadin/testbench/ElementQuery.java

```diff
@@ -48,33 +57,153 @@ public class ElementQuery<T extends TestBenchElement> {
     private static final int DEFAULT_WAIT_TIME_OUT_IN_SECONDS = 10;
 
     public static class AttributeMatch {
-        private final String name;
-        private final String operator;
-        private final String value;
-        private final boolean negate;
 
-        public AttributeMatch(String name, String operator, String value) {
-            this.name = name;
+        public enum Operator {
+            /**
+             * Attribute exists (with or without a value).
+             */
+            EXISTS(""),
+            /**
+             * Attribute value and given value match exactly.
+             */
+            MATCHES_EXACTLY("="),
+            /**
+             * Attribute value contains the given value.
+             */
+            CONTAINS("*="),
+            /**
+             * Attribute value contains a space-separated token
+             * that matches the given value.
+             */
+            CONTAINS_TOKEN("~="),
+            /**
+             * Attribute value begins with a space/hyphen-separated prefix
+             * that matches the given value.
+             * The prefix must be either the entire attribute value or
+             * the leading hyphen-separated segments of the attribute value.
+             */
+            CONTAINS_PREFIX("|="),
+            /**
+             * Attribute value begins with the given value.
+             */
+            BEGINS_WITH("^="),
```

Comment:
What about `STARTS_WITH`? It might be more familiar to Java developers.

Reply:
IMO, that was an unfortunate name choice if you want to maintain parallelism.
start → stop  ✅
begin → end  ✅
start → end  ❓

Reply:
I see. 
I just made a comparison with what we have in String class.
I'm fine also with begins. Feel free to choose what do you prefer

Reply:
Let's keep it the way it is.